src/interpreter.rs

```diff
@@ -1256,69 +1256,94 @@ impl<'source> Interpreter<'source> {
-            _ => bail!("internal error not a function"),
-        };
+        let fcns = fcns_rules.clone();
 
-        if args.len() != params.len() {
-            return Err(span.source.error(
-                span.line,
-                span.col,
-                format!(
-                    "mismatch in number of arguments. supplied {}, expected {}",
-                    params.len(),
-                    args.len()
-                )
-                .as_str(),
-            ));
-        }
+        let mut results: Vec<Value> = Vec::new();
+        for fcn_rule in fcns {
+            let (args, output_expr, bodies) = match fcn_rule {
+                Rule::Spec {
+                    head: RuleHead::Func { args, assign, .. },
+                    bodies,
+                    ..
+                } => (args, assign.as_ref().map(|a| &a.value), bodies),
+                _ => bail!("internal error not a function"),
+            };
+
+            if args.len() != params.len() {
+                return Err(span.source.error(
+                    span.line,
+                    span.col,
+                    format!(
+                        "mismatch in number of arguments. supplied {}, expected {}",
+                        params.len(),
+                        args.len()
+                    )
+                    .as_str(),
+                ));
+            }
 
-        let mut args_scope = Scope::new();
-        for (idx, a) in args.iter().enumerate() {
-            let a = match a {
-                Expr::Var(s) => s.text(),
-                _ => unimplemented!("destructuring function arguments"),
+            let mut args_scope = Scope::new();
+            for (idx, a) in args.iter().enumerate() {
+                let a = match a {
+                    Expr::Var(s) => s.text(),
+                    _ => unimplemented!("destructuring function arguments"),
+                };
+                //TODO: check call in params
+                args_scope.insert(a.to_string(), self.eval_expr(&params[idx])?);
+            }
+
+            let ctx = Context {
+                key_expr: None,
+                output_expr,
+                value: Value::new_set(),
             };
-            //TODO: check call in params
-            args_scope.insert(a.to_string(), self.eval_expr(&params[idx])?);
-        }
 
-        let ctx = Context {
-            key_expr: None,
-            output_expr,
-            value: Value::new_set(),
-        };
+            // Back up local variables of current function and empty
+            // the local variables of callee function.
+            let scopes = std::mem::take(&mut self.scopes);
 
-        // Back up local variables of current function and empty
-        // the local variables of callee function.
-        let scopes = std::mem::take(&mut self.scopes);
+            // Set the arguments scope.
+            self.scopes.push(args_scope);
+            let value = self.eval_rule_bodies(ctx, span, bodies)?;
```

Comment:
You could do something like this to accumulate errors
```suggestion
            let value = match self.eval_rule_bodies(ctx, span, bodies) {
               Ok(v) => v
               Err(e) => {
                    // If the rule produces an error, save the error.
                    errors.push(Err(e));
                    continue;
               }
            };
```